src/lib.rs

```diff
@@ -1076,6 +1140,39 @@ where
             let txid = maybe_await!(wallet.broadcast(tx))?;
             Ok(json!({ "txid": txid }))
         }
+        #[cfg(feature = "reserves")]
+        ProduceProof { msg } => {
+            let mut psbt = maybe_await!(wallet.create_proof(&msg))?;
+
+            let _finalized = wallet.sign(
+                &mut psbt,
+                SignOptions {
+                    trust_witness_utxo: true,
+                    ..Default::default()
+                },
+            )?;
+
+            let psbt_ser = serialize(&psbt);
+            let psbt_b64 = base64::encode(&psbt_ser);
+
+            Ok(json!({ "psbt": psbt , "psbt_base64" : psbt_b64}))
+        }
+        #[cfg(feature = "reserves")]
+        VerifyProof {
+            psbt,
+            msg,
+            confirmations,
+        } => {
+            let psbt = base64::decode(&psbt).unwrap();
+            let psbt: PartiallySignedTransaction = deserialize(&psbt).unwrap();
+            let current_height = wallet.client().get_height()?;
+            let _max_confirmation_height = current_height - confirmations.unwrap_or(6);
```

Comment:
If default is specified via struct opt then this can be simplified a little (and remove hard coded value here):
```suggestion
            let _max_confirmation_height = current_height - confirmations;
```